tests/containers/skopeo.pm

```diff
@@ -0,0 +1,119 @@
+# SUSE's openQA tests
+#
+# Copyright 2023-2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: skopeo
+# Summary: Test basic skopeo commands.
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';    # used in select_serial_terminal
+use utils 'zypper_call';    # used in zypper_call
+use version_utils 'package_version_cmp';
+use version_utils 'is_transactional';
+use transactional;
+
+sub run {
+    my ($self, $args) = @_;
+
+    # Ensure skopeo is installed
+    # people make a $packages array which is populated with the packages that
+    # they want installed.
+
+
+    ######### Set required variables
+
+    # Required packages
+    my $packages = 'skopeo jq';    # and add whatever else should be missing with push function
+
+    select_serial_terminal();    # Select most suitable text console
+
+    # Set a variable for my remote image
+    my $remote_image = 'docker://registry.suse.com/bci/bci-busybox:latest';
+    # Set a variable for my working directory
+    my $workdir = '/tmp/test';
+
+    ######### Ensure pre-requisites
+
+    # Install required packages
+    # Make logic to account for SLE-Micro environment with transactional-update
+    is_transactional ? trup_install($packages) : zypper_call("--quiet in $packages", timeout => 120);
+
+    # Skopeo version emits a reasonable output
+    # Verify that skopeo version is greater or equal to 1.12.0.
+    die('wrong skopeo version') unless (package_version_cmp('1.12.0', script_output('skopeo --version | cut -d" " -f3')) <= 0);
+    # skopeo --version | cut -d" " -f3 --> separates "skopeo version 1.12.0" output into and array and takes third value "1.12.0"
+
+    # Create test directory
+    script_run("mkdir /tmp/test");
+
+    ######### Inspect a docker repository
+
+    # assert_script_run --> runs the command and die unless it returns zero, indicating successful completion of $cmd
+    assert_script_run("skopeo inspect $remote_image",
+        fail_message => 'Failed to inspect remote image.');
+
+
+    ######### Pull the image into a directory
+
+
+    validate_script_output("skopeo copy $remote_image dir:$workdir",
+        sub { m/Writing manifest to image destination/ },
+        fail_message => 'Failed to copy image.');
+
+    # Unpacked contents must include a manifest and version
+    validate_script_output("file $workdir/manifest.json",
```

Comment:
```suggestion
    assert_script_output("stat $workdir/manifest.json", fail_message => "manifest.json not present");
```

Reply:
I would not rely on the error message output to determine if a file is there, but simply use the `stat` (or `cat`) program, which fails if the file doesn't exist.

Reply:
Relying on user-facing error messages might be flanky.

Reply:
Done! Much cleaner. I did not know the stat command.